sc/social/like/tests/test_canonicalurl_view.py

```diff
@@ -23,21 +24,31 @@ def setUp(self):
         self.portal = self.layer['portal']
         self.request = self.layer['request']
         enable_social_media_behavior()
+        self.setup_content()
+        api.portal.set_registry_record(
+            name='canonical_domain', value='https://example.org', interface=ISocialLikeSettings)
 
     def setup_content(self):
         with api.env.adopt_roles(['Manager']):
-            api.content.create(self.portal, type='News Item', id='foo')
-            api.content.create(self.portal, type='News Item', id='bar')
-            api.content.create(self.portal, type='News Item', id='baz')
+            obj = api.content.create(self.portal, type='News Item', id='foo')
+            api.content.transition(obj, 'publish')
+            obj = api.content.create(self.portal, type='News Item', id='bar')
+            api.content.transition(obj, 'publish')
+            obj = api.content.create(self.portal, type='News Item', id='baz')
+            api.content.transition(obj, 'publish')
 
         # simulate objects were create way long in the past
-        self.portal['foo'].creation_date = DateTime('2015/01/01')
-        self.portal['bar'].creation_date = DateTime('2016/01/01')
+        self.portal['foo'].effective_date = DateTime('2015/01/01')
         self.portal['foo'].reindexObject()
+        self.portal['bar'].effective_date = DateTime('2016/01/01')
         self.portal['bar'].reindexObject()
```

Comment:
we can add the date in the `api.content.create` call avoiding the need to reindex the object